test/hotspot/jtreg/compiler/intrinsics/math/TestMinMaxInlining.java

```diff
@@ -0,0 +1,161 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8307513
+ * @summary Test min and max IR inlining decisions
+ * @library /test/lib /
+ * @run driver compiler.intrinsics.math.TestMinMaxInlining
+ */
+
+package compiler.intrinsics.math;
+
+import compiler.lib.ir_framework.Argument;
+import compiler.lib.ir_framework.Arguments;
+import compiler.lib.ir_framework.Check;
+import compiler.lib.ir_framework.CompilePhase;
+import compiler.lib.ir_framework.IR;
+import compiler.lib.ir_framework.IRNode;
+import compiler.lib.ir_framework.Test;
+import compiler.lib.ir_framework.TestFramework;
+
+public class TestMinMaxInlining {
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    @Test
+    @Arguments(values = { Argument.NUMBER_MINUS_42, Argument.NUMBER_42 })
+    @IR(counts = { IRNode.MIN_I, "1" })
+    private static int testIntMin(int a, int b) {
+        return Math.min(a, b);
+    }
+
+    @Check(test = "testIntMin")
+    public static void checkTestIntMin(int result) {
+        if (result != -42) {
+            throw new RuntimeException("Incorrect result: " + result);
+        }
+    }
+
+    @Test
+    @Arguments(values = { Argument.NUMBER_MINUS_42, Argument.NUMBER_42 })
+    @IR(counts = { IRNode.MAX_I, "1" })
+    private static int testIntMax(int a, int b) {
+        return Math.max(a, b);
+    }
+
+    @Check(test = "testIntMax")
+    public static void checkTestIntMax(int result) {
+        if (result != 42) {
+            throw new RuntimeException("Incorrect result: " + result);
+        }
+    }
+
+    @Test
+    @Arguments(values = { Argument.NUMBER_MINUS_42, Argument.NUMBER_42 })
+    @IR(phase = { CompilePhase.BEFORE_MACRO_EXPANSION }, counts = { IRNode.MIN_L, "1" })
+    @IR(phase = { CompilePhase.AFTER_MACRO_EXPANSION }, counts = { IRNode.MIN_L, "0" })
+    private static long testLongMin(long a, long b) {
+        return Math.min(a, b);
+    }
+
+    @Check(test = "testLongMin")
+    public static void checkTestLongMin(long result) {
+        if (result != -42L) {
+            throw new RuntimeException("Incorrect result: " + result);
+        }
+    }
+
+    @Test
+    @Arguments(values = { Argument.NUMBER_MINUS_42, Argument.NUMBER_42 })
+    @IR(phase = { CompilePhase.BEFORE_MACRO_EXPANSION }, counts = { IRNode.MAX_L, "1" })
+    @IR(phase = { CompilePhase.AFTER_MACRO_EXPANSION }, counts = { IRNode.MAX_L, "0" })
+    private static long testLongMax(long a, long b) {
+        return Math.max(a, b);
+    }
+
+    @Check(test = "testLongMax")
+    public static void checkTestLongMax(long result) {
+        if (result != 42L) {
+            throw new RuntimeException("Incorrect result: " + result);
+        }
+    }
+
+    @Test
+    @Arguments(values = { Argument.NUMBER_MINUS_42, Argument.NUMBER_42 })
+    @IR(counts = { IRNode.MIN_F, "1" }, applyIfCPUFeatureOr = {"avx", "true"})
```

Comment:
Is this not supported by `asimd`? Same question for the other cases.

Reply:
Good point. I'll look into that aarch64 environments to see how things behave.